R/plots.R

```diff
@@ -125,6 +125,187 @@ plot_pca <- function(obj,
+#' @return a ggplot object
+#' @export
+plot_loadings <- function(obj, 
+  use_filtered = TRUE,
+  sample = NULL, #string please
+  PC = NULL, #apparently if u type in a string or an integer (corresponding to the PC), they're both ok
+  units = 'est_counts',
+  pc_count = NULL,
+  bool = FALSE,
+  absolute = TRUE, 
+  ...) {
+
+  stopifnot( is(obj, 'sleuth') )
+  #filtering?? doesn't work right now
+
+  # mat <- NULL
+  # if (use_filtered) {
+  #   mat <- spread_abundance_by(obj$obs_norm_filt, units)
+  # } else {
+  #   mat <- spread_abundance_by(obj$obs_norm, units)
+  # }
+  mat <- spread_abundance_by(obj$obs_norm, units)
+  
+  pca_calc <- prcomp(mat, scale = bool)
+
+  #sort of hack-y, may wish to fix
+  if (sample == '') {
+    sample <- NULL
+  }
+
+  loadings <- t(pca_calc$x)
+
+  #debugging
+  # print(sample)
+  # print(summary(loadings))
+  # print(pc_count)
+  # print("pc")
+  # print(PC)
+
+
+  executed <- FALSE
+  #given a sample
+  if (!is.null(sample)) {
+    executed <- TRUE
+    loadings <- pca_calc$x[sample,]
+    if (absolute) {
+      loadings <- abs(loadings)
+    }
+    loadings <- sort(loadings, decreasing = TRUE)
+    names <- names(loadings)
+  }
+
+  #given a PC, which samples contribute the most?
+  if (!executed) {
+    loadings <- pca_calc$x[,PC]
+    #print('reached here')
+    #print(loadings)
+    if (absolute) {
+      loadings <- abs(loadings)
+    }
+    loadings <- sort(loadings, decreasing = TRUE)
+    names <- names(loadings)
+  }
+
+
+  if (!is.null(pc_count)) {
+      loadings <- loadings[1:pc_count]
+      names <- names[1:pc_count]
+    } else {
+      loadings <- loadings[1:5]
+      names <- names[1:5]
+    }
+
+  dat <- data.frame(pc = names, loadings = loadings)
+  dat$pc <- factor(dat$pc, levels = unique(dat$pc))
+
+  p <- ggplot(dat, aes(x = pc, y = loadings)) 
+  p <- p + geom_bar(stat = "identity")
+  p <- p + xlab("Principal Components") + ylab("Contribution Scores")
+  if (!executed) {
+    p <- p + xlab("Transcripts")
+  }
+
+  #logistics of graph
+  #print(is.numeric(PC))
+
+  if (is.numeric(PC)) {
+    PC <- paste0("PC ", PC)
+  }
+
+  if (!is.null(sample)) {
+    p <- p + ggtitle(sample)
+  } else {
+    p <- p + ggtitle(PC)
+  }
+
+  p
+
+}
+
+#' Plot PC Variance
+#'
+#' Plot PC variances retained by percentage with option to compare specified PC
+#'
+#' @param obj a \code{sleuth} object
+#' @param use_filtered if TRUE, use filtered data. otherwise, use all data
+#' @param units either 'est_counts' or 'tpm'
+#' @param pca_number user input on how many PC to display, otherwise default is 5
+#' @param bool determines scaling
+#' @param PC_relative gives the option to compare subsequent principal components and their contributions
+#' @return a ggplot object
+#' @export
+plot_pc_variance <- function(obj, 
+  use_filtered = TRUE,
+  units = 'est_counts',
+  pca_number = NULL,
+  bool = FALSE,
+  PC_relative = NULL, #this is an integer
```

Comment:
lowercase please. again, if you are expecting a particular data type, please mention it in the documentation above